src/openms/source/FILTERING/CALIBRATION/InternalCalibration.cpp

```diff
@@ -344,13 +344,13 @@ namespace OpenMS
       tms.reserve(exp.size());
       // go through spectra and calibrate
       Size i(0), i_mslvl(0);
-      for (PeakMap::Iterator it = exp.begin(); it != exp.end(); ++it, ++i)
+      for (auto& spec : exp)
```

Comment:
this loop replacement is wrong.
note the continue statement.
also needs to increase i

Reply:
Currently not. Maybe in later c++ versions